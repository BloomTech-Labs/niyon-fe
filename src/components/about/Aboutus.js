import React from 'react';
import Button from '@material-ui/core/Button';
import { makeStyles } from '@material-ui/core/styles';
import './styles.scss';

const useStyles = makeStyles(theme => ({
  root: {
    display: 'flex',
  }
}));

function Aboutus() {
    const classes = useStyles();

  return (
        <div className="container">
          <h3>We are Niyon!
            </h3>
            </div>
  );
}
export default Aboutus;