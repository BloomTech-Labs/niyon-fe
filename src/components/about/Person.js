import React from 'react';
import { makeStyles } from '@material-ui/core/styles';
import './styles.scss';

const useStyles = makeStyles(theme => ({
  root: {
    display: 'flex',
  },
  person: {
    display: "flex",
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "center",
    textAlign: "center",
    margin: "32px",
    width: "250px"
},
personImg: {
    height: "150px",
    width: "150px",
    borderRadius: "50%",
    background: "#484848",
    overflow: "hidden",
    display: "flex",
    justifyContent: "center",
    alignItems: "center",
    boxShadow: "0 5px 25px rgba(0,0,0,0.2)",
    marginBottom: "10px"
},
h2: {
    margin: "0",
    marginTop: "10px"
},
p: {
    color: "#484848",
    margin: "0",
    marginTop: "10px"
}
}));



function Person({src, name, position}) {
    const classes = useStyles();

  return (
    <div className={classes.person}>
    <div className={classes.personImg}>
      <img src={src} alt="" height="100%" />
    </div>
    <h2>{name}</h2>
    <p>{position}</p>

  </div>
  );
}


export default Person;