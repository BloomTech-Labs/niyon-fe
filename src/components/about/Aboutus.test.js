import React from 'react';
import { shallow } from 'enzyme';
// import { createShallow } from '@material-ui/core/test-utils';
import toJSON from 'enzyme-to-json';
import Aboutus from './Aboutus';
import Person from './Person';
import findByTestAttr from '../../tests/utils';


const setUp = (props={}) => {
   const component = shallow(<Aboutus />);
   return component;
}

describe('<Aboutus /> component testing', () => {
   let component;
   beforeEach(() => {
     component = setUp();
   });

  it('should match the snapshot test correctly', () => {
      expect(toJSON(component)).toMatchSnapshot();
   })

 it('should render <AboutUs /> component correctly', () => {
      expect(component.length).toBe(1);
   });

 it('should render main container for <Aboutus /> component correctly', () => {
      const aboutWrapper = findByTestAttr(component, 'aboutUs-container');
      expect(aboutWrapper.length).toBe(1);
   });

 it('should render container for teams in <Aboutus /> component', () => {
      const teamContainer = findByTestAttr(component, 'team-container');
      expect(teamContainer.length).toBe(1);
   });

   describe('Testing team members rendering inside the component', () => {
    it('should render the team members on the component correctly', () => {         
         expect(component.find('.team')).toBeDefined();        
      });
    it('should render the <Person /> component correctly', () => {      
         expect(component.find('.team').at(0).find(Person).exists()).toBe(true);
         expect(component.find('.team').find(Person).length).toBe(7);
      });

      it('should match the  <Person /> snapshots "n" number of times', () => {         
           const result = component
                          .find('.team')
                          .find(Person)
                          .map( (node) => node );
            expect(toJSON(result)).toMatchSnapshot();           
      });     

   });

});
