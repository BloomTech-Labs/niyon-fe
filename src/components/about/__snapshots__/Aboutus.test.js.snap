// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Aboutus /> component testing should match the snapshor test corrrectly 1`] = `
<div
  className="container"
  data-test="aboutUs-container"
>
  <h1>
    Meet the Crew
  </h1>
  <div
    className="team"
    data-test="team-container"
  >
    <Person
      key="Carlos Banks"
      link="https://github.com/carlostbanks"
      name="Carlos Banks"
      position="Frontend Developer"
      src="/CB.jpeg"
    />
    <Person
      key="Kenneth Brook"
      link="https://github.com/wombat1972"
      name="Kenneth Brook"
      position="Frontend Developer"
      src="/KB.jpeg"
    />
    <Person
      key="Farjad Fazli"
      link="https://github.com/farjadfazli"
      name="Farjad Fazli"
      position="Frontend Developer"
      src="/FF.jpeg"
    />
    <Person
      key="Tim Mitzen"
      link="https://github.com/TimMitzen"
      name="Tim Mitzen"
      position="Backend Developer"
      src="/TM.jpeg"
    />
    <Person
      key="Joe Thompson"
      link="https://github.com/Joe-Thompson"
      name="Joe Thompson"
      position="Backend Developer"
      src="/JT.jpeg"
    />
    <Person
      key="Venky Yagatilee"
      link="https://github.com/vy3191"
      name="Venky Yagatilee"
      position="Front Developer"
      src="/VY.jpeg"
    />
    <Person
      key="Carlos Villalpando"
      link="https://github.com/CarlosVillalpandoJr"
      name="Carlos Villalpando"
      position="Project Manager"
      src="/CV.jpeg"
    />
  </div>
</div>
`;

exports[`<Aboutus /> component testing should match the snapshot test correctly 1`] = `
<div
  className="container"
  data-test="aboutUs-container"
>
  <h1>
    Meet the Crew
  </h1>
  <div
    className="team"
    data-test="team-container"
  >
    <Person
      key="Carlos Banks"
      link="https://github.com/carlostbanks"
      name="Carlos Banks"
      position="Frontend Developer"
      src="/CB.jpeg"
    />
    <Person
      key="Kenneth Brook"
      link="https://github.com/wombat1972"
      name="Kenneth Brook"
      position="Frontend Developer"
      src="/KB.jpeg"
    />
    <Person
      key="Farjad Fazli"
      link="https://github.com/farjadfazli"
      name="Farjad Fazli"
      position="Frontend Developer"
      src="/FF.jpeg"
    />
    <Person
      key="Tim Mitzen"
      link="https://github.com/TimMitzen"
      name="Tim Mitzen"
      position="Backend Developer"
      src="/TM.jpeg"
    />
    <Person
      key="Joe Thompson"
      link="https://github.com/Joe-Thompson"
      name="Joe Thompson"
      position="Backend Developer"
      src="/JT.jpeg"
    />
    <Person
      key="Venky Yagatilee"
      link="https://github.com/vy3191"
      name="Venky Yagatilee"
      position="Front Developer"
      src="/VY.jpeg"
    />
    <Person
      key="Carlos Villalpando"
      link="https://github.com/CarlosVillalpandoJr"
      name="Carlos Villalpando"
      position="Project Manager"
      src="/CV.jpeg"
    />
  </div>
</div>
`;
